test/unit/org/apache/cassandra/cql3/ast/Symbol.java

```diff
@@ -92,12 +95,35 @@ public AbstractType<?> type()
         return type;
     }
 
+    public AbstractType<?> rawType()
+    {
+        if (reversed) return ReversedType.getInstance(type);
+        return type;
+    }
+
     @Override
     public String name()
     {
         return symbol;
     }
 
+    /**
+     * Same as {@link #visit(Visitor)} but has the restriction that the return type may not be changed away from {@link Symbol}.
+     */
+    public Symbol visit(String type, Visitor v)
```

Comment:
nit: Could use the `Mutation.Kind` enum here instead of a string?

Reply:
currently I could but that then couples `Symbol` with `Mutation`.  The `String type` is just for better assert logic so rather not